generate/exportedtf/exportedtf.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exportedtf
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/madlambda/spells/errutil"
+	"github.com/mineiros-io/terramate"
+	"github.com/mineiros-io/terramate/config"
+	"github.com/mineiros-io/terramate/hcl"
+	"github.com/mineiros-io/terramate/hcl/eval"
+	"github.com/mineiros-io/terramate/stack"
+	"github.com/rs/zerolog/log"
+)
+
+// StackTf represents all exported terraform code for a stack,
+// mapping the exported code name to the actual Terraform code.
+type StackTf struct {
+	tfcode map[string]Body
+}
+
+// Body represents exported Terraform code from a single block.
+// Is contains parsed and evaluated code on it.
+type Body struct {
+	body []byte
```

Comment:
what about calling this field `content`?

Reply:
The type itself is already Body and this is just the private part of the body itself, so I don't see the point of calling each a different name :thinking: . I would then rename both to Content, but Body is shorter and it seems clear enough...or do you think body is confusing ? (I wrapped the []byte only to make it immutable, could also just pass around []byte).

Reply:
Tried to improve the names, not extremely happy with them overall